src/main/scala/com/johnsnowlabs/nlp/annotators/common/SentenceSplit.scala

```diff
@@ -18,15 +18,17 @@ package com.johnsnowlabs.nlp.annotators.common
 
 import com.johnsnowlabs.nlp.{Annotation, AnnotatorType}
 
+import scala.collection.Map
+
 /**
-  * structure representing a sentence and its boundaries
-  */
-case class Sentence(content: String, start: Int, end: Int, index: Int)
+ * structure representing a sentence and its boundaries
+ */
+case class Sentence(content: String, start: Int, end: Int, index: Int, metadata: Map[String, String] = null)
```

Comment:
Maybe is more usefull use Option[Map[String,String]]  it is a safer way to avoid nullPointerExceptions.
case class Sentence(content: String, start: Int, end: Int, index: Int, metadata: Map[String, String] = None)

Reply:
Is this better? (both the use of Option in SentenceSplit and TensorflowBert)

Reply:
Yes I think that is more scala idiomatic.